File: src/main/java/com/dh/ondot/core/config/AsyncProperties.java
package com.dh.ondot.core.config;

import lombok.Data;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.stereotype.Component;

@Data
@Component
@ConfigurationProperties(prefix = "async")
public class AsyncProperties {
    
    private final EventConfig event = new EventConfig();
    private final DiscordConfig discord = new DiscordConfig();

    @Data
    public static class EventConfig {
        private int corePoolSize = 4;
        private int maxPoolSize = 8;
        private int queueCapacity = 500;
    }

    @Data
    public static class DiscordConfig {
        private int corePoolSize = 2;
        private int maxPoolSize = 4;
        private int queueCapacity = 100;
    }
}

File: src/main/kotlin/com/dh/ondot/core/config/AsyncProperties.kt
package com.dh.ondot.core.config

import org.springframework.boot.context.properties.ConfigurationProperties

@ConfigurationProperties(prefix = "async")
data class AsyncProperties(
    val event: EventConfig = EventConfig(),
    val discord: DiscordConfig = DiscordConfig(),
) {

    data class EventConfig(
        val corePoolSize: Int = 4,
        val maxPoolSize: Int = 8,
        val queueCapacity: Int = 500,
    )

    data class DiscordConfig(
        val corePoolSize: Int = 2,
        val maxPoolSize: Int = 4,
        val queueCapacity: Int = 100,
    )
}
